packages/flutter/lib/src/widgets/two_dimensional_viewport.dart

```diff
@@ -0,0 +1,1216 @@
+  /// overriding the [paint] method.
+  ///
+  /// Returns null if there is no active child for the given [ChildVicinity].
+  RenderBox? getChildFor(ChildVicinity vicinity) {
+    return _children.containsKey(vicinity) ? _children[vicinity] : null;
+  }
+
+  @override
+  void attach(PipelineOwner owner) {
+    super.attach(owner);
+    _horizontalOffset.addListener(markNeedsLayout);
+    _verticalOffset.addListener(markNeedsLayout);
+    _delegate.addListener(_handleDelegateNotification);
+    for (final RenderBox child in _children.values) {
+      child.attach(owner);
+    }
+  }
+
+  @override
+  void detach() {
+    super.detach();
+    _horizontalOffset.removeListener(markNeedsLayout);
+    _verticalOffset.removeListener(markNeedsLayout);
+    _delegate.removeListener(_handleDelegateNotification);
+    for (final ChildVicinity cellIndex in _children.keys) {
+      _children[cellIndex]!.detach();
+    }
+  }
+
+  @override
+  void redepthChildren() {
+    for (final RenderBox child in _children.values) {
+      child.redepthChildren();
+    }
+  }
+
+  @override
+  void visitChildren(RenderObjectVisitor visitor) {
+    _childrenInPaintOrder.forEach(visitor);
+  }
+
+  @override
+  void visitChildrenForSemantics(RenderObjectVisitor visitor) {
+    // Only children that are visible should be visited, and they must be in
+    // paint order.
+    childrenInPaintOrder
+      .where((RenderBox child) => parentDataOf(child).isVisible)
+      .forEach(visitor);
+  }
+
+  @override
+  List<DiagnosticsNode> debugDescribeChildren() {
+    final List<DiagnosticsNode> debugChildren = <DiagnosticsNode>[
+      ..._children.keys.map<DiagnosticsNode>((ChildVicinity index) {
+        return _children[index]!.toDiagnosticsNode(name: index.toString());
+      })
+    ];
+    return debugChildren;
+  }
+
+  @override
+  Size computeDryLayout(BoxConstraints constraints) {
+    debugCheckHasBoundedAxis(Axis.vertical, constraints);
+    debugCheckHasBoundedAxis(Axis.horizontal, constraints);
+    return constraints.biggest;
+  }
+
+  @override
+  bool hitTestChildren(BoxHitTestResult result, {required Offset position}) {
+    for (final RenderBox child in _children.values) {
+      final TwoDimensionalViewportParentData childParentData = parentDataOf(child);
+      final Rect childRect = childParentData.paintOffset & child.size;
+      if (childRect.contains(position)) {
+        result.addWithPaintOffset(
+          offset: childParentData.paintOffset,
+          position: position,
+          hitTest: (BoxHitTestResult result, Offset transformed) {
+            assert(transformed == position - childParentData.paintOffset);
+            return child.hitTest(result, position: transformed);
+          },
+        );
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /// The dimensions of the viewport.
+  ///
+  /// This [Size] represents the width and height of the visible area.
+  Size get viewportDimension {
```

Comment:
I added it for convenience. Looking at the table implementation, I thought it was better for readability. `viewportDimension` is commonly used in ScrollMetrics and SliverConstraints, etc, so I thought adding this would be a nice through line.